nussl/separation/nmf_mfcc.py

```diff
@@ -0,0 +1,215 @@
+            distance_measure (str): The type of distance measure to use in NMF - euclidean or divergence.
+            num_iterations (int): The number of iterations to go through in NMF.
+            random_seed (int): The seed to use in the numpy random generator
+            kmeans_kwargs (dict): The kwargs for KMeans parameters.
+            convert_to_mono (bool): Given a stereo signal, convert to mono.
+            mask_type (object): A soft or binary mask object used for the source separation.
+            mfcc_range (int/list/tuple): The range of mfcc to use for clustering.
+            n_mfcc (int): The max number of mfccs to use.
+
+        Attributes:
+            input_audio_signal (object): An Audio Signal object of the input audio signal.
+            clusterer (object): A KMeans object for clustering the templates and activations.
+            signal_stft (np.matrix): The stft data for the input audio signal.
+            labeled_templates (np.array): A Numpy array containing the labeled templates columns
+                                               from the templates matrix for a particular source.
+            sources (np.array): A Numpy array containing the list of Audio Signal objects for each source.
+            masks (np.array): A Numpy array containing the lists of Binary Mask objects for each channel.
+
+        """
+    def __init__(self, input_audio_signal, num_sources, num_templates=50, distance_measure='euclidean',
+                 num_iterations=50, random_seed=None, kmeans_kwargs=None, convert_to_mono=False,
+                 mask_type=mask_separation_base.MaskSeparationBase.BINARY_MASK, mfcc_range=(1, 14), n_mfcc=20):
+        super(NMF_MFCC, self).__init__(input_audio_signal=input_audio_signal, mask_type=mask_type)
+
+        self.num_sources = num_sources
+        self.num_templates = num_templates
+        self.distance_measure = distance_measure
+        self.num_iterations = num_iterations
+        self.random_seed = random_seed
+        self.kmeans_kwargs = kmeans_kwargs
+        self.convert_to_mono = convert_to_mono
+        self.n_mfcc = n_mfcc
+        self.mask_type = mask_type
+
+        self.signal_stft = None
+        self.input_audio_signal = input_audio_signal
+        self.labeled_templates = None
+        self.sources = []
+        self.masks = []
+
+        # Convert the stereo signal to mono if indicated
+        if self.convert_to_mono:
+            self.input_audio_signal.to_mono(overwrite=True, remove_channels=False)
+
+        # Set the MFCC range
+        if isinstance(mfcc_range, int) and mfcc_range < n_mfcc:
+            self.mfcc_start, self.mfcc_end = 1, mfcc_range
+        elif isinstance(mfcc_range, (tuple, list)) and len(mfcc_range) == 2:
+            self.mfcc_start, self.mfcc_end = mfcc_range[0], mfcc_range[1]
+        else:
+            raise ValueError('mfcc_range is not set correctly! Must be a tuple or list with min and max, or int (max)')
```

Comment:
Hmmm.... I'm not sure I would understand what this error message was telling me...